torch/package/_directory_reader.py

```diff
@@ -44,9 +32,10 @@ def get_record(self, name):
             return f.read()
 
     def get_storage_from_record(self, name, numel, dtype):
-        storage = _dtype_to_storage[dtype]
         filename = f"{self.directory}/{name}"
-        return _HasStorage(storage.from_file(filename=filename, size=numel))
+        nbytes = torch._utils._element_size(dtype) * numel
+        storage = cast(Storage, torch.ByteStorage)
+        return _HasStorage(storage.from_file(filename=filename, size=nbytes))
```

Comment:
It maybe marginally safer to rename the kwarg here to nbytes (from size), to help other people catch if they were feeding numel rather than bytes.